test: workqueue: add code description

yingmingx opened the pull request:
1. Modify testcase, add ztest case in test_workq_start_before_submit().
2. Add doxygen comments for details of workqueue.

Signed-off-by: Ying ming <mingx.ying@intel.com>

LeiW000 commented on tests/kernel/workq/work_queue_api/src/main.c:
```diff
@@ -135,19 +164,46 @@ void test_process_work_items_fifo(void)
+ * -# Then compare results.
+ *
+ * Expected Test Result:
+ * - The measured time results are in a very small range.
```
Here, it's better to specify that range.

enjiamai replied:
@yingmingx It seems like the "check point" was not been pointed out in your test procedure, it's a little ambiguous here. 
You could declare it like this:  "Then check if the X-time is within Y and Z".    or  something likes that.

enjiamai commented on tests/kernel/workq/work_queue_api/src/main.c:
```diff
@@ -67,13 +67,42 @@ void common_work_handler(struct k_work *unused)
+ * sync_sema with initial count 0.
+ * -# Handler function gives semaphore, then we wait for that semaphore
+ * from the test function body.
+ * -# If semaphore was obtained successfully, test passed.
```
Recommend:  "Check" if the semaphore was obtained successfully.

yingmingx replied:
> About the descriptions of procedure, could you please clarify the check point as I comment? Thanks!

Thank you for the reminder. Indeed, I need to write the test steps in this format.

enjiamai commented on tests/kernel/workq/work_queue_api/src/main.c:
```diff
@@ -176,12 +232,41 @@ void test_sched_delayed_work_item(void)
+ * struct k_work_q.
+ * -#  Define and initialize maximum possible real number of the
+ * workqueues available according to the stack size.
+ * -#  Test defines and initializes max number of the workqueues
```
Recommend: "Check" would be better here.

enjiamai commented on tests/kernel/workq/work_queue_api/src/main.c:
```diff
@@ -640,13 +725,40 @@ static void ttriggered_work_cancel(const void *data)
+ * Test Procedure:
+ * -# Call the k_work_q_start() to create a work queue which has it's own
+ * thread that processes the work items in the queue.
+ *
```
no check point here?

yingmingx replied:
I spilt the PR into 2 commits. The first commit use ztest to check this case and add code descriptions in the second commit.

enjiamai commented on tests/kernel/workq/work_queue_api/src/main.c:
```diff
@@ -832,20 +944,49 @@ static void work_handler_resubmit(struct k_work *w)
+ * @brief Test resubmitting work item in callback
  * @details
- * - That test uses sync_sema semaphore with initial count 0.
- * - That test verifies that it is possible during execution of the handler
```
These two sentence looks not like "test procedures", how do you think remove "The test" here?

enjiamai reviewed (changes requested):
About the descriptions of procedure, could you please clarify the check point as I comment? Thanks!

yingmingx commented:
> About the descriptions of procedure, could you please clarify the check point as I comment? Thanks!

Thank you for the reminder. Indeed, I need to write the test steps in this format.

Commit:
test: workqueue: modify testcase

The thread name is used to judge whether the thread of
user-defined workqueue is created successfully.

Signed-off-by: Ying ming <mingx.ying@intel.com>

LeiW000 commented:
@yingmingx "buildkite/zephyr" check failed. Please fix it.

Commit:
test: workqueue: add code description

Add testcase descriptions for test_work_item_supplied_with_func(),
test_sched_delayed_work_item(), test_workqueue_max_number(),
test_workq_start_before_submit(), test_work_submit_handler()

Signed-off-by: Ying ming <mingx.ying@intel.com>

LeiW000 commented:
@nashif , @andrewboie , @andyross , does anyone help to merge the PR?
